docs/contributor/arch/016-transform-filter-support-for-telemetry-pipelines.md

```diff
@@ -0,0 +1,117 @@
+# 16. Transform and Filter support for OpenTelemetry based Pipelines
+
+Date: 2024-10-10
+
+## Status
+
+Proposed
+
+## Context
+
+In the default setup of metric and trace pipelines, users currently cannot filter, transform, or enrich data before it is sent to the backend. To provide more flexibility, users should be able to filter data based on specific conditions and apply transformations or enrichment to the data. This solution must ensure a consistent approach across all OpenTelemetry (OTel) pipelines.
+
+## Decision
+
+We will implement a consolidated solution in the OpenTelemetry Collector (OTel Collector) using a single Filter and Transform Processor. This processor will use the OpenTelemetry Transformation and Transport Language (OTTL) to handle both filtering and transformation tasks. Users will be able to configure the processor as they need; and only a subset of OTTL functions will be supported, focusing on the most common and impactful use cases.
+
+Example configuration MetricPipeline:
+    
+```yaml
+apiVersion: telemetry.kyma-project.io/v1alpha1
+kind: MetricPipeline
+metadata:
+  name: metricpipeline-sample
+spec:
+  transform:
+    conditions:
+      - type == METRIC_DATA_TYPE_SUM
+    statements:
+      - set(description, "Sum")
+  filter:
+    metric:
+        - type == METRIC_DATA_TYPE_NONE
+    datapoint:
+        - metric.name == "k8s.pod.phase" and value_int == 4
+  input:
+      istio:
+        enabled: true
+      prometheus:
+        enabled: false
+  output:
+    otlp:
+      endpoint:
+        value: ingest-otlp.services.sap.hana.ondemand.com:443
+```
+
+Example configuration TracePipeline:
+
+```yaml
+apiVersion: telemetry.kyma-project.io/v1alpha1
+kind: TracePipeline
+metadata:
+  name: tracepipeline-sample
+spec:
+  transform:
+    statements:
+        - set(status.code, 1) where attributes["http.path"] == "/health"
+  filter:
+    span:
+      - IsMatch(resource.attributes["k8s.pod.name"], "my-pod-name.*")
+  output:
+    otlp:
+      endpoint:
+        value: ingest-otlp.services.sap.hana.ondemand.com:443
+      protocol: grpc
+```
+### Solution
+
+The OTTL library offers functions to validate, filter, and transform data by parsing and executing expressions. To ensure robustness and stability, we will limit the available filter and transform functions to a carefully selected subset. This subset will be curated based on common use cases, ensuring essential functionality without overcomplicating the configuration process, the OTTL library provides two categories of functions, the editor functions and the converters.
```

Comment:
there are two flavours of functions:
editor functions and converters. The ADR goes into more detail for the editor functions, but does not go into detail for converters. will the be 
- fully supported
- partially supported
- not supported?